Data/Aeson/Types/FromJSON.hs

```diff
@@ -1541,12 +1541,15 @@ instance FromJSONKey Float where
         _           -> Scientific.toRealFloat <$> parseScientificText t
 
 instance (FromJSON a, Integral a) => FromJSON (Ratio a) where
-    parseJSON = withObject "Rational" $ \obj -> do
-        numerator <- obj .: "numerator"
-        denominator <- obj .: "denominator"
-        if denominator == 0
-        then fail "Ratio denominator was 0"
-        else pure $ numerator % denominator
+    parseJSON (Number x) = pure (realToFrac x)
```

Comment:
This is security risk, think `1e9999999`, see how `Integer` is parsed, e.g.

Reply:
Sorry for the delay. I sort of forgot about this PR. I updated it with check analogous to one for Integer 

I also noticed that there's unused import